src/libpriv/rpmostree-postprocess.c

```diff
@@ -212,12 +352,37 @@ do_kernel_prep (int            rootfs_dfd,
                              cancellable, error))
     return FALSE;
 
-  if (!rpmostree_finalize_kernel (rootfs_dfd, bootdir, kver,
-                                  kernel_path,
+  /* We always tell rpmostree_finalize_kernel() to skip /boot, since we'll do a
+   * full hardlink pass if needed after that for the kernel + bootloader data.
+   */
+  if (!rpmostree_finalize_kernel (rootfs_dfd, bootdir, kver, kernel_path,
                                   &initramfs_tmpf,
+                                  RPMOSTREE_FINALIZE_KERNEL_USRLIB_OSTREEBOOT,
                                   cancellable, error))
     return FALSE;
 
+  /* We always ensure this exists as a mountpoint */
+  if (!glnx_ensure_dir (rootfs_dfd, "boot", 0755, error))
+    return FALSE;
+
+  /* If the boot location includes /boot, we also need to copy /usr/lib/ostree-boot there */
+  switch (boot_location)
```

Comment:
This can just be `if (boot_location == RPMOSTREE_POSTPROCESS_BOOT_LOCATION_BOTH)` right?

Reply:
Yeah, but I usually like doing an explicit `switch` over enums so one gets warnings when adding a new case (and down the line we'll probably add a `LOCATION_MODULES` or so).